source/mbed_trace.c

```diff
@@ -454,8 +454,12 @@ void mbed_vtracef(uint8_t dlevel, const char* grp, const char *fmt, va_list ap)
 
 end:
     if ( m_trace.mutex_release_f ) {
-        while (m_trace.mutex_lock_count > 0) {
-            m_trace.mutex_lock_count--;
+        // Store the mutex lock count to temp variable so that it won't get
```

Comment:
This might be a good time to add some kind of comments why the unlocking loop exists at all. Every time I have looked that code it has given a impression that something must be very, very wrong here. 
+1 for functional part.